src/utilities/paramsFinder.ts

```diff
@@ -1,11 +1,44 @@
+import { Param } from '@/types'
+import { setParamValue } from '@/utilities/params'
 import { stringHasValue } from '@/utilities/string'
 
-export function findParamValues(url: string, path: string, paramName: string): string[] {
+export function getParamValuesFromUrl(url: string, path: string, paramName: string): string[] {
+  const regexPattern = getParamRegexPattern(path, paramName)
+
+  return getCaptureGroups(url, regexPattern)
+}
+
+export type ParamReplace = {
+  name: string,
+  params?: Param[],
+  values?: unknown[],
+}
+
+export function setParamValuesOnUrl(path: string, paramReplace?: ParamReplace): string {
+  if (!paramReplace) {
+    return path
+  }
+
+  const { name } = { ...paramReplace }
+  const params = paramReplace.params ?? []
+  const values = paramReplace.values ?? []
```

Comment:
I think this is the same and more concise yeah? If not just ignore. 
```suggestion
  const { name, params = [], values = [] } = { ...paramReplace }
```

Reply:
much better, thank you 🙏